app/models/questionnaire.rb

```diff
@@ -1,6 +1,6 @@
 class Questionnaire < OpenStruct
   def self.find_by!(args)
-    questionnaire_data = QuestionnaireData.find_by!(args)
+    questionnaire_data = SummativeQuestionnaire.find_by(args) ? SummativeQuestionnaire.find_by(args):QuestionnaireData.find_by(args)
     questionnaire_data.build_questionnaire
   end
 
```

Comment:
Having seen a separate SummativeQuestionnaire I then didn't expect to see it being used in the Questionnaire model.  I may need you to draw me a picture of what you're doing here.